spatialpandas/geometry/base.py

```diff
@@ -624,13 +739,343 @@ def fillna(self, value=None, method=None, limit=None):
+            with the supplied bounds
+        """
+        raise NotImplementedError()
+
+
+class _CoordinateIndexer(object):
+    def __init__(self, obj, parent=None):
+        self._obj = obj
+        self._parent = parent
+
+    def __getitem__(self, key):
+        obj = self._obj
+        xs, ys = key
+        # Handle xs and ys as scalar numeric values
+        if type(xs) is not slice:
+            xs = slice(xs, xs)
+        if type(ys) is not slice:
+            ys = slice(ys, ys)
+
+        if xs.step is not None or ys.step is not None:
+            raise ValueError(
+                "Slice step not supported. The cx indexer uses slices to represent "
+                "intervals in continuous coordinate space, and a slice step has no "
+                "clear interpretation in this context."
+            )
+        xmin, ymin, xmax, ymax = obj.sindex.total_bounds
+        x0, y0, x1, y1 = (
+            xs.start if xs.start is not None else xmin,
+            ys.start if ys.start is not None else ymin,
+            xs.stop if xs.stop is not None else xmax,
+            ys.stop if ys.stop is not None else ymax,
+        )
+
+        # Handle inverted bounds
+        if x1 < x0:
+            x0, x1 = x1, x0
+        if y1 < y0:
+            y0, y1 = y1, y0
+
+        covers_inds, overlaps_inds = obj.sindex.covers_overlaps((x0, y0, x1, y1))
+        overlaps_inds_mask = obj.intersects_bounds((x0, y0, x1, y1), overlaps_inds)
+        selected_inds = np.sort(
+            np.concatenate([covers_inds, overlaps_inds[overlaps_inds_mask]])
+        )
+        if self._parent is not None:
+            return self._parent.iloc[selected_inds]
+        else:
+            return obj[selected_inds]
+
+
+@jit(nopython=True, nogil=True)
+def _lexographic_lt0(a1, a2):
+    """
+    Compare two 1D numpy arrays lexographically
+    Parameters
+    ----------
+    a1: ndarray
+        1D numpy array
+    a2: ndarray
+        1D numpy array
+
+    Returns
+    -------
+    comparison:
+        True if a1 < a2, False otherwise
+    """
+    for e1, e2 in zip(a1, a2):
+        if e1 < e2:
+            return True
+        elif e1 > e2:
+            return False
+    return len(a1) < len(a2)
+
+
+def _lexographic_lt(a1, a2):
+    if a1.dtype != np.object and a1.dtype != np.object:
+        # a1 and a2 primitive
+        return _lexographic_lt0(a1, a2)
+    elif a1.dtype == np.object and a1.dtype == np.object:
+        # a1 and a2 object, process recursively
+        for e1, e2 in zip(a1, a2):
+            if _lexographic_lt(e1, e2):
+                return True
+            elif _lexographic_lt(e2, e1):
+                return False
+        return len(a1) < len(a2)
+    elif a1.dtype != np.object:
+        # a2 is object array, a1 primitive
+        return True
+    else:
+        # a1 is object array, a2 primitive
+        return False
+
+
+@ngjit
+def _take_offsets(inds, all_offsets, values):
```

Comment:
Can you explain a bit why this was needed? The `take()` method of a pyarrow Array does not what you want?

(want to understand if there is some functionality missing in pyarrow that we should add upstream)

Reply:
ugh, I thought I had tried pyarrow's `take` but I think I must have tried plain `__getitem__`, and since this didn't seem to support array indexing I assumed I'd have to write it myself.  I'll make this update in https://github.com/jonmmease/spatialpandas/pull/5 since I made some other changes to `take` over there. Thanks!